tools/deno

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env -S deno run --unstable --allow-read --allow-write --allow-env --no-check
+
+/* eslint no-console: "off" */
+/* eslint no-process-exit: "off" */
+
+import { existsSync } from "https://deno.land/std@0.90.0/fs/exists.ts";
+
+import { DefaultTestFileLoader } from "../out/common/framework/file_loader.js";
+import { Logger } from "../out/common/framework/logging/logger.js";
+import { parseQuery } from "../out/common/framework/query/parseQuery.js";
+import { assert, unreachable } from "../out/common/framework/util/util.js";
+
+function usage(rc) {
+  console.log("Usage:");
+  console.log("  tools/deno [OPTIONS...] QUERIES...");
+  console.log("  tools/deno 'unittests:*' 'webgpu:buffers,*'");
+  console.log("Options:");
+  console.log("  --verbose     Print result/log of every test as it runs.");
+  console.log("  --debug       Include debug messages in logging.");
+  console.log("  --print-json  Print the complete result JSON in the output.");
+  return Deno.exit(rc);
+}
+
+function printResults(results) {
+  for (const [name, r] of results) {
+    console.log(`[${r.status}] ${name} (${r.timems}ms). Log:`);
+    if (r.logs) {
+      for (const l of r.logs) {
+        console.log("  - " + l.toJSON().replace(/\n/g, "\n    "));
+      }
+    }
+  }
+}
+
+if (!existsSync("src/common/runtime/cmdline.ts")) {
+  console.log("Must be run from repository root");
+  usage(1);
+}
+
+const badComment = `/// <reference types="@webgpu/types" />`;
+const navigatorGpuPath = "./out/webgpu/util/navigator_gpu.js";
+let navigatorGpuFile = await Deno.readTextFile(navigatorGpuPath);
+navigatorGpuFile = navigatorGpuFile.replace(
+  badComment,
+  " ".repeat(badComment.length),
+);
+await Deno.writeTextFile(navigatorGpuPath, navigatorGpuFile);
+
+let verbose = false;
+let trace = false;
+let debug = false;
+let ignoreFile = undefined;
+let printJSON = false;
+const queries = [];
+for (const a of Deno.args) {
+  if (a.startsWith("-")) {
+    if (a === "--verbose") {
+      verbose = true;
+    } else if (a === "--trace") {
+      trace = true;
+    } else if (a === "--debug") {
+      debug = true;
+    } else if (a === "--print-json") {
+      printJSON = true;
+    } else if (a.startsWith("--ignore-file=")) {
+      ignoreFile = a.slice("--ignore-file=".length);
```

Comment:
FYI, I think soon we might be able to replace this with something integrated into the CTS, as I think we will soon teach it to understand "expectations". That's not ready yet though; we can let you know.
cc @austinEng

Reply:
Awesome. Looking forward to that :-)